cmd/skywire-cli/commands/log/root.go

```diff
@@ -141,56 +155,74 @@ var logCmd = &cobra.Command{
-					defer httpC.CloseIdleConnections()
-					defer wg.Done()
-
-					deleteOnError := false
-					if _, err := os.ReadDir(key); err != nil {
-						if err := os.Mkdir(key, 0750); err != nil {
-							log.Errorf("Unable to create directory for visor %s", key)
-							return
+						deleteOnError := false
+						if _, err := os.ReadDir(key); err != nil {
+							if err := os.Mkdir(key, 0750); err != nil {
+								log.Errorf("Unable to create directory for visor %s", key)
+								return
+							}
+							deleteOnError = true
 						}
-						deleteOnError = true
-					}
-					// health check before downloading anything else
-					// delete that folder if the health check fails
-					err = download(ctx, log, httpC, "health", "health.json", key, maxFileSize)
-					if err != nil {
-						if deleteOnErrors {
-							if deleteOnError {
-								bulkFolders = append(bulkFolders, key)
+						// health check before downloading anything else
+						// delete that folder if the health check fails
+						err = download(ctx, log, httpC, "health", "health.json", key, maxFileSize)
+						if err != nil {
+							if deleteOnErrors {
+								if deleteOnError {
+									bulkFolders = append(bulkFolders, key)
+								}
+								return
 							}
-							return
 						}
-					}
-					if !logOnly {
-						download(ctx, log, httpC, "node-info.json", "node-info.json", key, maxFileSize) //nolint
-					}
-					if !surveyOnly {
-						for i := 0; i <= duration; i++ {
-							date := time.Now().AddDate(0, 0, -i).UTC().Format("2006-01-02")
-							download(ctx, log, httpC, "transport_logs/"+date+".csv", date+".csv", key, maxFileSize) //nolint
+						if !logOnly {
+							download(ctx, log, httpC, "node-info.json", "node-info.json", key, maxFileSize) //nolint
 						}
+						if !surveyOnly {
+							for i := 0; i <= duration; i++ {
+								date := time.Now().AddDate(0, 0, -i).UTC().Format("2006-01-02")
+								download(ctx, log, httpC, date+".csv", date+".csv", key, maxFileSize) //nolint
+							}
+						}
+					}(v.PubKey, &wg)
+					batchSize--
+					if batchSize == 0 {
+						time.Sleep(15 * time.Second)
+						batchSize = 50
 					}
-				}(v.PubKey, &wg)
-				batchSize--
-				if batchSize == 0 {
-					time.Sleep(15 * time.Second)
-					batchSize = 50
+				}
+				//omit the filters if a file was specified
+				if fetchFile != "" {
+					wg.Add(1)
+					go func(key string, wg *sync.WaitGroup) {
+						httpC := http.Client{Transport: dmsghttp.MakeHTTPTransport(ctx, dmsgC), Timeout: 10 * time.Second}
+						defer httpC.CloseIdleConnections()
+						defer wg.Done()
+						if _, err := os.ReadDir(key); err != nil {
+							if err := os.Mkdir(key, 0750); err != nil {
+								log.Errorf("Unable to create directory for visor %s", key)
+								return
+							}
+						}
+						_ = download(ctx, log, httpC, fetchFile, fetchFile, key, maxFileSize) //nolint
+					}(v.PubKey, &wg)
```

Comment:
The default behavior of `skywire-cli log` is designed to be used for rewards.

Thus, the minimum version filter is used to keep from collecting surveys for visors which are not eligible or which have become ineligible.

IF in the future there is ever a reason to collect a certain file or files en-mass from visors or from some subset of visors, and this file is not there explicitly for the purpose of rewards or it's not a file we are otherwise currently generating or collecting, `fetchFile` is provided to facilitate that.

It leaves a lot of room for the users to take advantage of the existing framework for whatever creative purpose they come up with, and adds flexibility for development going forward along the same lines.